lib/pry/input_completer.rb

```diff
@@ -239,4 +227,30 @@ def build_path(input)
     end
     return path, input
   end
+  
+    private
+
+    def self.fetch_obj_space_collection
+      (self.obj_spase_mutex ||= Mutex.new).synchronize do
+        count = ObjectSpace.each_object(Module).count
+        return self.obj_space_collection if count == self.obj_space_count
+        self.obj_space_collection  = []
+        buffer = {}
+        self.obj_space_count = ObjectSpace.each_object(Module) do |m|
+          begin
+            name = m.name.to_s
+          rescue Pry::RescuableException
+            name = ""
+          end
+          next if name != "IRB::Context" and
+              /^(IRB|SLex|RubyLex|RubyToken)/ =~ name
+
+          # jruby doesn't always provide #instance_methods() on each
+          # object.
+          # this is the fastest way to make this collection uniq
+          m.instance_methods(false).each{|method| buffer[method] = nil }
+        end
+        self.obj_space_collection = buffer.keys.map(&:to_s).sort
+      end
```

Comment:
shouldn't you call `.count` here? otherwise obj_space_count is going to be an Enumerator. i don't think the cache will ever be invalidated because an enumerator cannot be equal to a number(which `count` local holds).